snippets/code_to_review1.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+
+def compCostFunction(estim_y, true_y):
+    E = estim_y - true_y
+    C = (1 / 2 * m) * np.sum(E ** 2)
+    return C
+
+def myFunction(x, y):
+    # this checks whether the x and y have the same number of samples
+    assert isinstance(x, np.ndarray), "Only works for arrays"
+    assert isinstance(y, np.ndarray), "Only works for arrays"
+    return x.shape[0] == y.shape[0]
```

Comment:
If I were to review this, and depending of the author's Python experience, I'd suggest using a try-except block
```
try:
    return x.shape[0] == y.shape[0]
except AttributeError:
    raise AttribureError("x and y should probably be numpy arrays")
```

Reply:
Totally agree with @tlestang 

`assert` statements in non-test code should have a very limited use. Nonetheless, that code doesn't handle error conditions, just fires them up (which should be acceptable sometimes when the function is a _utility_ function, never to be used alone but as part of a larger modularised algorithm. 

However, in that case, naming should be different (e.g. adding an _underscore_ at the beginning to mark it as a _private_ not-to-import function!

Reply:
Both good points, I'll leave it as a code smell, but again added to notes for a reviewer.